forge-game/src/main/java/forge/game/ability/effects/VoteEffect.java

```diff
@@ -143,28 +141,35 @@ public void resolve(final SpellAbility sa) {
                 }
             }
         } else {
-            List<String> subAbs = Lists.newArrayList();
-            final List<Object> mostVotes = getMostVotes(votes);
-            if (sa.hasParam("Tied") && mostVotes.size() > 1) {
-                subAbs.add(sa.getParam("Tied"));
-            } else if (sa.hasParam("VoteSubAbility")) {
-                host.addRemembered(mostVotes);
-                subAbs.add(sa.getParam("VoteSubAbility"));
+            List<SpellAbility> subAbs = Lists.newArrayList();
+            if (sa.hasParam("StoreVoteNum") && sa.hasParam("Choices")) {
+                for (final Object type : voteType) {
+                    SpellAbility subAb = (SpellAbility) type;
+                    subAb.setSVar("VoteNum", "Number$" + votes.get(type).size());
+                    subAbs.add((SpellAbility)type);
+                }
             } else {
-                for (Object type : mostVotes) {
-                    subAbs.add(sa.getParam("Vote" + type.toString()));
+                final List<Object> mostVotes = getMostVotes(votes);
+                if (sa.hasAdditionalAbility("VoteTiedAbility") && mostVotes.size() > 1) {
+                    subAbs.add(sa.getAdditionalAbility("VoteTiedAbility"));
+                } else if (sa.hasAdditionalAbility("VoteSubAbility")) {
+                    host.addRemembered(mostVotes);
+                    subAbs.add(sa.getAdditionalAbility("VoteSubAbility"));
+                } else {
+                    for (Object type : mostVotes) {
+                        subAbs.add((SpellAbility)type);
```

Comment:
```
Game-0 > java.lang.ClassCastException: class forge.game.player.Player cannot be cast to class forge.game.spellability.SpellAbility (forge.game.player.Player and forge.game.spellability.SpellAbility are in unnamed module of loader 'app')
	at forge.game.ability.effects.VoteEffect.resolve(VoteEffect.java:160)
```

Reply:
should be fixed

Reply:
```
Game-0 > java.lang.ClassCastException: class forge.game.card.Card cannot be cast to class forge.game.spellability.SpellAbility (forge.game.card.Card and forge.game.spellability.SpellAbility are in unnamed module of loader 'app')
	at forge.game.ability.effects.VoteEffect.resolve(VoteEffect.java:160)
```

maybe the logic can be cleaned up further so you don't have to mix all these `hasParam` checks multiple times